hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -274,6 +281,21 @@ protected void syncHoodieTable(String tableName, boolean useRealtimeInputFormat,
     }
   }
 
+  private boolean isAlreadySynced(String tableName) {
+    return syncClient.getLastCommitTimeSynced(tableName)
+        .map(lastCommit -> {
+          Option<String> lastCompletion =
+              syncClient.getLastCommitCompletionTimeSynced(tableName);
```

Comment:
shouldn't we check the last completion time synced with the current latest completion time on timeline?

Reply:
Yes, that is what `getCommitsTimelineAfter` is doing based on my understanding.

Reply:
Not really, you even passed the instant time instead of the completion time there.

Reply:
What do you mean? `lastCompletion` is completion time 😅

Reply:
Not sure what we want to check here: `TimelineUtils.getCommitsTimelineAfter` is designed to catch up some "hollow" instants before V8, so it takes a instant time and a completion time.

But here you pass around the last synced instant time and last synced completion time, it does not make any sense.

Reply:
I am open to changes so long as they are still allowing us to exit early in the case where there are no commits to sync so if there is a better or simpler way you would prefer, please post a snippet to speed up the process.

Reply:
Like i mentioned: "check the last completion time synced with the current latest completion time on timeline"